### DISCUSSION TITLE
Nuke's Image UIViewRepresentable

### DISCUSSION THREAD

**Discussion by jose-pay** [kean/Nuke#658] [General]
Hi,

some thoughts about Nuke 11:

- I was wondering why Nuke also has a view named `Image`, which is the same name as SwiftUI's`Image`. I end up having to namespace SwiftUI's image when NukeUI library is imported. Was there any particular reason naming it `Image` as well instead of giving it another name? This leads to confusion. 
- The `LazyImage`'s completion closure result in an `if let / else if` syntax that doesn't look like typical SwiftUI. Did you consider the option to use closures like [`AsyncImage`](https://developer.apple.com/documentation/swiftui/asyncimage) does?  
- I see Nuke as one product, but I now have to use three different imports. I can understand that maybe not everyone uses SwiftUI, but having to add a NukeExtensions import for 2 files seems a bit of a stretch. 

Thanks for a great library and look forward to hear your thoughts on this.

**Comment by kean** [comment]
Hi,

> Was there any particular reason naming it Image as well instead of giving it another name?

I was going to replace it but couldn't come up with a good name. Any suggestions?

> Did you consider the option to use closures like [AsyncImage](https://developer.apple.com/documentation/swiftui/asyncimage) does?

Do you mean `content` and `placeholder`? Yeah, I guess I could add it as a convenience. There is an `error` state though. Not sure having three closures is better than one with a switch.

> having to add a NukeExtensions import for 2 files seems a bit of a stretch.

Many folks only use SwiftUI. NukeUI has public typealiases for the primary Nuke types, so you typically only need one import: `import NukeUI`. NukeUI isn't just for SwiftUI though. It has `LazyImageView` for UIKit/AppKit that I recommend using instead of `NukeExtensions` – these are no longer the main API for working with image views and are close to being deprecated.

**Comment by kean** [comment]
> Was there any particular reason naming it Image as well instead of giving it another name?

I'm considering removing Gifu integration (making it optional) and extracting video playback as well from NukeUI. With this change, there is going to be no need for `NukeUI.Image` anymore.

**Reply by jose-pay** [reply]
> There is an error state though. Not sure having three closures is better than one with a switch.

Ah ok, I see now, the current approach is basically the same as AsyncImage:
https://developer.apple.com/documentation/swiftui/asyncimage/init(url:scale:transaction:content:)

> With this change, there is going to be no need for NukeUI.Image anymore.

Sounds good to me.

Thanks.
